smif/model/scenario_model.py

```diff
@@ -66,11 +70,136 @@ def add_data(self, data, timesteps):
         >>> elec_scenario.add_data(data, timesteps)
         """
         self.timesteps = timesteps
-        assert isinstance(data, ndarray)
+        assert isinstance(data, np.ndarray)
         self._data = data
 
     def simulate(self, timestep, data=None):
         """Returns the scenario data
         """
         time_index = self.timesteps.index(timestep)
         return {self.name: {self.model_outputs.names[0]: self._data[time_index]}}
+
+
+class ScenarioModelBuilder(object):
+
+    def __init__(self, name):
+        self.scenario = ScenarioModel(name)
+        self.logger = getLogger(__name__)
+
+        self.region_register = get_region_register()
+        self.interval_register = get_interval_register()
+
+    def construct(self, scenario_config, data, timesteps):
+        """Build the complete and populated ScenarioModel
+
+        Assumes that a ScenarioModel can only have one output
+
+        Arguments
+        ---------
+        scenario_config : dict
+        data : list
+        timesteps : list
+        """
+        name = scenario_config['name']
+
+        spatial = scenario_config['spatial_resolution']
+        temporal = scenario_config['temporal_resolution']
+
+        spatial_res = self.region_register.get_entry(spatial)
+        temporal_res = self.interval_register.get_entry(temporal)
+
+        self.scenario.add_output(name,
+                                 spatial_res,
+                                 temporal_res,
+                                 scenario_config['units'])
+
+        array_data = self._data_list_to_array(name, data,
+                                              timesteps,
+                                              spatial_res,
+                                              temporal_res)
+
+        self.scenario.add_data(array_data, timesteps)
+
+    def finish(self):
+        """Return the built ScenarioModel
+        """
+        return self.scenario
+
+    def _data_list_to_array(self, param, observations, timestep_names,
```

Comment:
This looks generally useful - note to come back to this if it should live in a reader or be tucked into the data layer somehow.